src/utils/cardUtilityFunctions.js

```diff
@@ -122,3 +124,403 @@ export const getUpdatedCardSize = oldSize => {
+  const { cardVariables } = updatedCard;
+
+  // check for title variables and replace them
+  const titleVariables = getVariables(title);
+  if (titleVariables) {
+    const updatedTitle = replaceVariables(titleVariables, cardVariables, title);
+    updatedCard.title = updatedTitle || title;
+  }
+
+  const { columns, thresholds } = updatedCard.content;
+  columns.forEach((column, i) => {
+    const { linkTemplate } = column;
+    if (linkTemplate) {
+      const { href, displayValue } = linkTemplate;
+      // Check for variables in the hrefs
+      const hrefVariables = getVariables(href);
+      if (hrefVariables) {
+        const updatedHref = replaceVariables(hrefVariables, cardVariables, href);
+        updatedCard.content.columns[i].linkTemplate.href = updatedHref;
+      }
+      // Check for variables in the display values
+      const displayValueVariables = getVariables(displayValue);
+      if (displayValueVariables) {
+        const updatedDisplayValue = replaceVariables(
+          displayValueVariables,
+          cardVariables,
+          displayValue
+        );
+        updatedCard.content.columns[i].linkTemplate.displayValue = updatedDisplayValue;
+      }
+    }
+  });
+  if (thresholds) {
+    thresholds.forEach((threshold, x) => {
+      const { label, severityLabel, value } = threshold;
+      // Check if there are variables in the threshold labels
+      const thresholdLabelVariables = getVariables(label);
+      if (thresholdLabelVariables) {
+        const updatedLabel = replaceVariables(thresholdLabelVariables, cardVariables, label);
+        updatedCard.content.thresholds[x].label = updatedLabel;
+      }
+      // Check if there are variables in the threshold severity labels
+      const severityLabelVariables = getVariables(severityLabel);
+      if (severityLabelVariables) {
+        const updatedSeverityLabel = replaceVariables(
+          severityLabelVariables,
+          cardVariables,
+          severityLabel
+        );
+        updatedCard.content.thresholds[x].severityLabel = updatedSeverityLabel;
+      }
+      // Check if there are variables in the threshold values
+      const thresholdValueVariables = getVariables(value);
+      if (thresholdValueVariables) {
+        const updatedValue = replaceVariables(thresholdValueVariables, cardVariables, value);
+        updatedCard.content.thresholds[x].value = updatedValue;
+      }
+    });
+  }
+
+  return updatedCard;
+};
+
+export const handleVariables = (cardType, title, content, values, card) => {
+  const updatedCard = {
+    title,
+    content,
+    values,
+    ...card,
+  };
+  if (!updatedCard.cardVariables) {
+    return updatedCard;
+  }
+  const { cardVariables } = updatedCard;
+
+  // check for title variables on all cards and replace them
+  const titleVariables = getVariables(title);
+  if (titleVariables) {
+    const updatedTitle = replaceVariables(titleVariables, cardVariables, title);
+    updatedCard.title = updatedTitle || title;
+  }
```

Comment:
This is reused in many different functions, would it make sense to pull out into it's own function to avoid repetition?